FreeRTOS/FreeRTOS
# Use new QEMU test project to improve stream/message buffer tests

Description
-----------
Improve the robustness of the stream and message buffer tests - required various minor updates to StreamBufferDemo.c and MessageBufferDemo.c to account for variations in the execution pattern related to how far through a tick period tasks started executing and the priority of other test tasks.

Also check in the FreeRTOS/Demo/CORTEX_LM3S6965_QEMU project used to first investigate then update the StreamBufferDemo.c and Message BufferDemo.c files.  The QEMU project is useful for recording execution traces, using thread aware FreeRTOS plug-ins, and testing different GCC versions.

Test Steps
-----------
Run the CORTEX_LM3S6965_QEMU project as will be documented on the FreeRTOS.org site.  Leave the project running and observe the QEMU output.  Note the demo will eventually fail due to too many tick interrupts occurring between the the task that tests the stream buffer trigger level unblocking and subsequently executing - that failure does not occur on real hardware.

Related Issue
-----------
None.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
